reconcile/utils/oc.py

```diff
@@ -108,8 +108,12 @@ def wrapper(*args, **kwargs):
                 return result
 
             function_name = f'{function.__module__}.{function.__qualname__}'
-            ignore_reconcile_time = \
+            ignore_reconcile_time_annotation_present = \
                 annotations.get('qontract.ignore_reconcile_time') == 'true'
+            is_e2e_test = running_state.get('e2e_test', None)
```

Comment:
`RunningState()` has not `get()` method. You can use `getattr(running_state, 'e2e_test', None)`, but I'd rather just set a default for any `RunningState()` attribute instead of creating multiple execution paths here.

`RunningState()` is super prone to race conditions. Having a default will make it safer.

You can just add to `RunningState()`:

```python
    def __getattr__(self, item):
        """
        Default value for attributes not explicetly created is None.
        """
```

With that, if something breakes because the attibute is `None`, that something has to check the attribute first. ATM, the only thing that could break is the `reconcile_time.labels()` call bellow, but Histograms have no issues with `label=None` as argument, so I'd not touch it.

Reply:
so in fact this change is sufficient and nothing else is required!